Add nodev,noexec,nosuid options to /var/log and /var/log/audit.

adelton opened the pull request:
#### Description:

- Add nodev,noexec,nosuid options to /var/log and /var/log/audit.

#### Rationale:

- Restricting operations which should never be needed on /var/log/*.

adelton commented:
Similar to https://github.com/ComplianceAsCode/content/pull/4542, we'd need to know the severity, the references, whether to list older RHEL versions, and if to add this to the ospp.profile files right away.

yuumasato commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_noexec/rule.yml:
```diff
@@ -0,0 +1,24 @@
+    {{{ describe_mount(option="noexec", part="/var/log") }}}
+
+rationale: |-
+    Allowing users to execute binaries from world-writable directories
```
Are `/var` and  `/var/log` world-writable?

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml:
```diff
@@ -0,0 +1,26 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
```suggestion
prodtype: rhel6,rhel7,rhel8,fedora,ol8
```

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml:
```diff
@@ -0,0 +1,26 @@
+    The only legitimate location for device files is the <tt>/dev</tt> directory
+    located on the root partition. The only exception to this is chroot jails.
+
+severity: unknown
```
```suggestion
severity: medium
```

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml:
```diff
@@ -0,0 +1,26 @@
+identifiers:
+    cce@rhel8: 81095-2
+
+references:
```
Add nist references to ``CM-7`` and ``MP-2``

CM-7 -- operate system with least privilege. Doesn't need nodev to operate

MP-2  -- multi media policy around removable drives

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml:
```diff
@@ -0,0 +1,26 @@
+severity: unknown
+
+identifiers:
+    cce@rhel8: 81095-2
```
Please add RHEL 6 and RHEL 7 CCEs

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml:
```diff
@@ -0,0 +1,24 @@
+identifiers:
+    cce@rhel8: 82975-4
+
+references:
```
NIST reference to ``CM-7``

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml:
```diff
@@ -0,0 +1,24 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
```suggestion
prodtype: rhel6,rhel7,rhel8,fedora,ol8
```

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml:
```diff
@@ -0,0 +1,24 @@
+severity: unknown
+
+identifiers:
+    cce@rhel8: 82975-4
```
rhel6 and rhel7 CCE

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml:
```diff
@@ -0,0 +1,24 @@
+    such as <tt>/var/log/audit</tt> should never be necessary in normal operation and
+    can expose the system to potential compromise.
+
+severity: unknown
```
```suggestion
severity: medium
```

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml:
```diff
@@ -0,0 +1,24 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
```suggestion
prodtype: rhel6,rhel7,rhel8,fedora,ol8
```

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml:
```diff
@@ -0,0 +1,24 @@
+severity: unknown
+
+identifiers:
+    cce@rhel8: 82921-8
```
rhel6, rhel7 CCEs

shawndwells commented on linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml:
```diff
@@ -0,0 +1,24 @@
+    The presence of SUID and SGID executables should be tightly controlled. Users
+    should not be able to execute SUID or SGID binaries from temporary storage partitions.
+
+severity: unknown
```
```suggestion
severity: medium
```

adelton commented:
In c8a8764ada5671a901d1cff66c2042b83806e478, I'v now rebased on master, plus I've:
- fixed the description of the purpose of these directories;
- added the medium severity;
- added the rhel8/templates/csv/mount_options.csv entries;
- added references `nist: CM-7` and `nist-csf: PR.IP-1,PR.PT-2,PR.PT-3` to all, plus `nist: MP-2` to the nodev ones.

I did not add the RHEL 6 and RHEL 7 stuff. Let's backport it there once the RHEL 8 stuff is done and added to the ospp profiles.

yuumasato commented:
LGTM.
@adelton Please, revolve the conflicts.

adelton commented:
Rebased on master -> 5520796fb915300a144d61aa4a7072e0528ee1c3.

yuumasato commented:
@shawndwells We can add these rules to RHEL6 and RHEL7 in later PR.
Ack, thank you @adelton.
